Library/Homebrew/brew.sh

```diff
@@ -294,6 +299,19 @@ EOS
 }
 check-run-command-as-root
 
+check-prefix-is-not-tmpdir() {
+  if [[ "${HOMEBREW_PREFIX}" = "${HOMEBREW_TEMP}"* ]]
```

Comment:
> check against the system temp rather than just HOMEBREW_TEMP

I might be misunderstanding, but: won't this leave open the case where they set both their prefix and their temp to something weird, e.g., `$HOME`? We want to stop all situations where Homebrew is both installing to and using the same prefix as a temporary directory, not just `/private/tmp`.

Reply:
@woodruffw Ok so is it fine to e.g. have the prefix in `/tmp` if `HOMEBREW_TEMP` is set elsewhere?

Reply:
Yep, exactly.